### PR TITLE
enable J3/geminal jastrow for PBCs

### PR DESCRIPTION
**Pull request by willwheelera**
This PR enables J3 to be used in periodic calculations.

Changes made: 
- new function to construct supercell for gamma point orbital evaluation
- J3 generates the appropriate supercell by default and creates PBC orbitals object
- PBC orbital evaluation only compute wrap phase if using more than gamma point (to ensure gamma point values are real)
- example periodic_j3.py

### CODE REVIEW

**Comment by lkwagner** [comment]
Can you explain this code? 
```
    def __init__(self, cell, mo_coeff=None, kpts=None):
         self.iscomplex = True
         self._cell = cell.original_cell
         self.S = cell.S
         self.Lprim = self._cell.lattice_vectors()

         self._kpts = [0, 0, 0] if kpts is None else kpts
         if mo_coeff is None:
             print("Orbitals initialized without MOs. Can only evaluate AOs")
         else:
             nelec_per_kpt = [np.asarray([m.shape[1] for m in mo]) for mo in mo_coeff]
             self.param_split = [
                 np.cumsum(nelec_per_kpt[spin])
                 for spin in [0, 1]
             ]
             self.parm_names = ["_alpha", "_beta"]
             self.parameters = {
                 "mo_coeff_alpha": gpu.cp.asarray(np.concatenate(mo_coeff[0], axis=1)),
                 "mo_coeff_beta": gpu.cp.asarray(np.concatenate(mo_coeff[1], axis=1)),
             }
             ```
I don't understand why mo_coeff is allowed to be None and then there is a printout and nothing happens. If it's intended that mo_coeff can be passed as None then there shouldn't be a bare printout like that, and if it's not then why is it an option?

**Comment by lkwagner** [comment]
Thanks for the PR! Please see the review, pull master to get the fix for the bad import statement I implemented today, and also please include a test in the PR.

File: pyqmc/orbitals.py [hunk 1]
```diff
@@ -333,7 +327,17 @@ def aos(self, eval_str, configs, mask=None):
                 Ls=self.Ls,
             )
         )
-        ao = gpu.cp.einsum("k...,k...a->k...a", wrap_phase, ao)
+        # coordinate, dimension
+        if np.abs(self._kpts).sum() > 1e-9:
```
[the remark this answers is not in the document]
Comment (willwheelera) [reply pyqmc/orbitals.py:334]: if only evaluating at gamma point, wrap_phase is not needed, and we don't want to make the AOs complex. I added a comment.

File: pyqmc/orbitals.py [hunk 2]
```diff
@@ -213,23 +213,26 @@ class PBCOrbitalEvaluatorKpoints:
...
-            np.cumsum(nelec_per_kpt[spin])
-            for spin in [0, 1]
-        ]
-        self.parm_names = ["_alpha", "_beta"]
-        self.parameters = {
-            "mo_coeff_alpha": gpu.cp.asarray(np.concatenate(mo_coeff[0], axis=1)),
-            "mo_coeff_beta": gpu.cp.asarray(np.concatenate(mo_coeff[1], axis=1)),
-        }
+        if mo_coeff is None:
+            print("Orbitals initialized without MOs. Can only evaluate AOs")
```
[the remark this answers is not in the document]
Comment (willwheelera) [reply pyqmc/orbitals.py]: Agreed, I added a docstring

File: pyqmc/orbitals.py [hunk 3]
```diff
@@ -213,23 +213,26 @@ class PBCOrbitalEvaluatorKpoints:
...
-        self.param_split = [
-            np.cumsum(nelec_per_kpt[spin])
-            for spin in [0, 1]
-        ]
-        self.parm_names = ["_alpha", "_beta"]
-        self.parameters = {
-            "mo_coeff_alpha": gpu.cp.asarray(np.concatenate(mo_coeff[0], axis=1)),
-            "mo_coeff_beta": gpu.cp.asarray(np.concatenate(mo_coeff[1], axis=1)),
-        }
+        if mo_coeff is None:
```
[the remark this answers is not in the document]
Comment (willwheelera) [reply pyqmc/orbitals.py]: yes, I made a note in the docstring about it

File: pyqmc/j3.py [hunk 4]
```diff
@@ -125,3 +131,31 @@ def testvalue(self, e, epos, mask=None):
             optimize=self.optimize,
         )
         return gpu.asnumpy(gpu.cp.exp((new_val.T - curr_val).T)), new_ao_val
+
+
+def make_pbc_supercell_for_gamma_aos(scell, S=None, **kwargs):
```
[the remark this answers is not in the document]
Comment (willwheelera) [reply pyqmc/j3.py:136]: k2gamma seems to work on scf objects, not on cell objects. I didn't find a pyscf function that accomplishes this. Functions they have for creating supercells only seem to work for diagonal supercell matrices (ncopies in each direction), while we are able to change the shape or orientation of the supercell.

File: pyqmc/orbitals.py [hunk 5]
```diff
@@ -333,7 +330,17 @@ def aos(self, eval_str, configs, mask=None):
                 Ls=self.Ls,
             )
         )
-        ao = gpu.cp.einsum("k...,k...a->k...a", wrap_phase, ao)
+        # If gamma-point only, do not compute wrap_phase, keep AOs real-valued
```
Comment (lkwagner) [inline pyqmc/orbitals.py:333]: I still don't quite understand this. Why is this sum being evaluated every time? 
If we are returning real values, is self.complex still True? 

The decision should be made in __init__ not in aos().

Also I'm a little worried about summing self._kpts(). Wouldn't that be zero also if we have an even mesh? This is not the same as checking whether we are gamma. Do we also know that the AOs are always real for gamma? What about when there is spin-orbit?

**Pull request merged**